### PR TITLE
Add introductory message for leaders new to Hackbot

### LINKED ISSUE #35: Add introductory message to Hackbot's leader check-in functionality
If we haven't run a check-in with a given user before, we should have some sort of introductory message ("Hi! I'm Hackbot, it looks like we haven't done this sort of thing before. <insert summary of check-in functionality.").

Thanks for the suggestion @vaibhavyadaram!

### PR DESCRIPTION
**Pull request by paked**
Fix #35

### CODE REVIEW

**Comment by zachlatta** [comment]
We talked about this in-person, but I'll reiterate here for the sake of consistency. This change won't work because `CheckIn` objects are only created when a leader responds to @hackbot's check-in prompt and has had a meeting (see existing logic in the check-in conversation to see what I mean).

**Comment by paked** [comment]
Blocking on a review of #54 by @zachlatta to implement the above changes.

File: app/models/hackbot/conversations/check_in.rb [hunk 1]
```diff
@@ -10,13 +10,24 @@ def self.should_start?(event, _team)
...
-        msg_channel "Hey #{first_name}! Did you have a club meeting this week?"
+        leader_info = leader(event)
+        first_name = leader_info.name.split(' ').first
+
+        if first_check_in?
+          msg_channel "Hi #{first_name}! I'm Hackbot - Hack Club's friendly robotic helper. "\
+            "Right now I'm reaching out to you so I can collect some data about your club's "\
+            'activity in the past week so that the team can improve your experience working '\
+            'with them :) '\
+            'So, first question: Did you have a club meeting this week?'
```
Comment (zachlatta) [inline app/models/hackbot/conversations/check_in.rb]: Couple things here.

- Do you want this to be one giant block of text? Or broken up with new lines? Currently a giant block of text.
- Let's use emoji instead of text smiley faces with Hackbot.

Copy can also be improved, that sentence uses the word "so" a lot right now.

> Hey Harrison! I'm Hackbot, Hack Club's friendly robot helper.
>
> I'll be reaching out to you every week, typically on Fridays, to check in and see how your club's doing. I'll be sharing everything with the team, so they'll be in the loop every step of the way 🙂.
>
> To start, did you have a club meeting this week?

File: app/models/hackbot/conversations/help.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module Hackbot
+  module Conversations
+    class Help < Hackbot::Conversations::Channel
+      HELP_MENU = "Hi! I'm Hack Club's friendly robotic helper. I don't do "\
+                  "much right now, but if you're a club leader you've "\
+                  'probably received some messages from me about the '\
+                  'attendance of your club.'\
+                  "\n\n"\
```
Comment (zachlatta) [inline app/models/hackbot/conversations/help.rb]: Probably a bit more idiomatic to have one newline at the end of line 7 and then the second one on this line (line 8).

That way it's clear where the line breaks are.

File: app/models/hackbot/conversations/help.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module Hackbot
+  module Conversations
+    class Help < Hackbot::Conversations::Channel
+      HELP_MENU = "Hi! I'm Hack Club's friendly robotic helper. I don't do "\
```
Comment (zachlatta) [inline app/models/hackbot/conversations/help.rb]: "friendly robotic helper" -> "friendly robot helper"

File: app/models/hackbot/conversations/help.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                  "much right now, but if you're a club leader you've "\
+                  'probably received some messages from me about the '\
+                  'attendance of your club.'\
+                  "\n\n"\
+                  "If you're interested in contributing to my codebase check "\
+                  "out https://github.com/hackclub/api/issues. I'd love to "\
+                  'learn some new tricks from you!'.freeze
```
Comment (zachlatta) [inline app/models/hackbot/conversations/help.rb]: Love this.

File: app/models/hackbot/conversations/help.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+      def self.should_start?(event, team)
+        Hackbot::Conversations::Mention.should_start?(event, team) &&
+          event[:text].include?('help')
```
Comment (zachlatta) [inline app/models/hackbot/conversations/help.rb]: This is really clever, but probably simpler to just do `event[:text] =~ /hackbot help/`.

This might have some unintended side effects (ex. what happens if a user mentions "help hackbot"?).

File: db/schema.rb [hunk 6]
```diff
@@ -23,6 +23,7 @@
     t.text     "notes"
     t.datetime "created_at",   null: false
     t.datetime "updated_at",   null: false
+    t.text     "did_happen"
```
Comment (zachlatta) [inline db/schema.rb]: Is this intentional?

File: app/models/check_in.rb [hunk 7]
```diff
@@ -2,5 +2,6 @@ class CheckIn < ApplicationRecord
   belongs_to :club
   belongs_to :leader
 
-  validates :club, :leader, :meeting_date, :attendance, presence: true
+  validates :club, :leader, :meeting_date, :attendance,
+            presence: true
```
Comment (zachlatta) [inline app/models/check_in.rb]: Was this change intentional?

**Review by zachlatta — COMMENTED** [review]
Looking real good, much better than before. Just some small comments before merging.

**Comment by zachlatta** [comment]
Ah snap, meant to leave those comments saying "request changes".
